File: 백준/Silver/5800. 성적 통계/성적 통계.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int k = Integer.parseInt(br.readLine());
        for(int t=1;t<=k;t++){
            String[] input = br.readLine().split(" ");
            int n = Integer.parseInt(input[0]);
            int[] scores = new int[n];

            for(int i=0;i<n;i++){
                scores[i] = Integer.parseInt(input[i+1]);
            }
            Arrays.sort(scores);
            int maxScore = scores[n - 1];
            int minScore = scores[0];
            int largestGap = 0;
            for (int i = 0; i < n - 1; i++) {
                largestGap = Math.max(largestGap, scores[i + 1] - scores[i]);
            }
         
            System.out.println("Class " + t);
            System.out.println("Max " + maxScore + ", Min " + minScore + ", Largest gap " + largestGap);
        }
    }
}

File: 백준/Silver/5800. 성적 통계/성적 통계.py
import sys
input = sys.stdin.readline

k = int(input())
for _ in range(k):
    arr = list(map(int,input().split()))
    n = arr[0]
    scores = arr[1:]
    max_score = max(scores)
    min_score = min(scores)
    score = sorted(scores,reverse=True)
    gap = 0
    for i in range(n-1):
        gap = max(gap, score[i] - score[i + 1])

    print(f"Class {_+1}")
    print(f"Max {max_score}, Min {min_score}, Largest gap {gap}")